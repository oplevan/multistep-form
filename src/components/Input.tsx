import * as React from "react";

import { cn } from "../utils/cn";

export interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {}

export default function Input({ className, type, ...props }: InputProps) {
  return (
    <input
      type={type}
      className={cn(
        "flex h-8 w-full rounded-md border p-2 text-sm focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 placeholder:text-slate-400",
        type === "date" && "block",
        className
      )}
      {...props}
    />
  );
}
