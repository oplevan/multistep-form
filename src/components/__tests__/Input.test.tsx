import { screen, render } from "@testing-library/react";
import Input from "../Input";

describe("<Input /> component", () => {
  test("should render correctly", () => {
    const { container } = render(<Input />);
    expect(container).toMatchSnapshot();
  });

  test("should have correct class name & attributes", () => {
    render(
      <Input
        custom-attribute="test"
        className="test"
        type="text"
      />
    );
    const input = screen.getByRole("textbox");
    expect(input).toBeInTheDocument();
    expect(input).toHaveClass("test");
    expect(input).toHaveAttribute("type", "text");
    expect(input).toHaveAttribute("custom-attribute", "test");
  });
});
