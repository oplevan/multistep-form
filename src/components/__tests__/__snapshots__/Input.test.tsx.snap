// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> component should render correctly 1`] = `
<div>
  <input
    class="flex h-8 w-full rounded-md border p-2 text-sm focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 placeholder:text-slate-400"
  />
</div>
`;

exports[`Input should render correctly 1`] = `
<div>
  <input
    class="flex h-8 w-full rounded-md border p-2 text-sm focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 placeholder:text-slate-400"
  />
</div>
`;
